Source/DynamicFeatureRegistry.swift

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the MIT license. See LICENSE file in https://github.com/yahoo/Override for terms.
+
+import Foundation
+
+public enum DynamicFeatureRegistryError: Error {
+    case staticFeatureAlreadyExists
+}
+
+/// DynamicFeatureRegistry a subclass of FeatureRegistry which provides an interface for using
+/// a collection of dynamic features instead of just pre-defined ones. When a feature is added,
+/// there is an attempt to apply an override state from the underlying data FeatureStore.
+@objc open class DynamicFeatureRegistry: FeatureRegistry {
+
+    /// A local store of the current dynamic features
+    private var dynamicFeatures: [String : AnyFeature] = [:]
+
+    override var features: [LabeledItem] {
+
+        let dynamicLabeledItems = dynamicFeatures.values.map {
+            LabeledFeatureItem(label: $0.key, feature: $0)
+        }
```

Comment:
This might be a bug – the order is lost here when using `values` of a dictionary. It would be easy enough to keep a list `[LabeledItem]` here with the side effect of making the feature lookup method slower

Reply:
Just to clarify, are you suggesting I simply change the type of the `dynamicLabeledItems` property to an array? 

Interestingly enough, the documentation mentions that the order of key-value pairs in a dictionary is stable between mutations but is otherwise unpredictable.

Now that I'm thinking about order, as they are added may not be the most desirable, especially if its being done dynamically. Maybe I'll provide a user customizable sort block instead.

Reply:
Oh yeah, it's complicated for sure. I think alpha ordering is perfectly reasonable.